File: Coding/patterns.py
def one():
    for i in range(0, 4):
        for j in range(0, 4):
            print("*", end="")
        print()


def two():
    for i in range(0, 4):
        for j in range(0, 4):
            if j not in (0, 3) and i not in (0, 3):
                print(end=" ")
            else:
                print("*", end="")
        print()


def three():
    for i in range(0, 4):
        for j in range(0, i):
            print(end=" ")
        for j in range(0, 4):
            print("*", end="")
        print()


def four():
    for i in range(0, 4):
        for j in range(0, 40):
            print("*", end="")
        print()


def five():
    for i in range(0, 4):
        for j in range(0, 40):
            if i not in (0, 3) and j not in (0, 39):
                print(end=" ")
            else:
                print("*", end="")
        print()


def six():
    for i in range(0, 10):
        for j in range(0, i):
            print(end=" ")
        for j in range(0, 20):
            print("*", end="")
        print()


def seven():
    for i in range(0, 10):
        for j in range(0, 10 - i):
            print(end=" ")
        for j in range(0, 10):
            print("*", end="")
        print()


def eight():
    for i in range(0, 4):
        for j in range(0, i + 1):
            print("*", end="")
        print()


def nine():
    for i in range(0, 5):
        for j in range(0, 5 - i):
            print(end=" ")
        for j in range(0, 2 * i + 1):
            print("*", end="")
        print()


def ten():
    for i in range(0, 5):
        for j in range(0, 5 - i):
            print(end=" ")
        for j in range(0, 2 * i + 1):
            if i == 4:
                print("*", end="")
            else:
                if j == 0 or j == 2 * i:
                    print("*", end="")
                else:
                    print(end=" ")
        print()


def eleven():
    for i in range(0, 9):
        for j in range(0, i):
            print(end=" ")
        for j in range(0, 2 * (9 - i) - 1):
            print("*", end="")
        print()


def twelve():
    for i in range(0, 12):
        for j in range(0, i):
            print(end=" ")
        for j in range(0, 2 * (11 - i) + 1):
            if i == 0 or j == 0 or j == 2 * (11 - i):
                print("*", end="")
            else:
                print(end=" ")
        print()


def thirteen():
    for i in range(0, 10):
        if i < 5:
            for j in range(0, i):
                print("*", end="")
        else:
            for j in range(0, 10 - i):
                print("*", end="")
        print()


def fourteen():
    for i in range(0, 10):
        if i < 5:
            for j in range(0, 10 - i):
                print(end=" ")
            for j in range(0, i):
                print("*", end="")
        else:
            for j in range(0, i):
                print(end=" ")
            for j in range(0, 10 - i):
                print("*", end="")
        print()


def fifteen():
    for i in range(0, 10):
        if i < 5:
            for j in range(0, 10 - i):
                print(end=" ")
            for j in range(0, 2 * i + 1):
                print("*", end="")
        else:
            for j in range(0, i+1):
                print(end=" ")
            for j in range(0, 20 - (2 * i) - 1):
                print("*", end="")
        print()


fifteen()

File: Coding/patterns.cpp
#include <bits/stdc++.h>
using namespace std;
void one()
{
    for (int i = 0; i < 4; i++)
    {
        for (int j = 0; j < 4; j++)
        {
            cout << "*";
        }
        cout << endl;
    }
}
void two()
{
    for (int i = 0; i < 4; i++)
    {
        for (int j = 0; j < 4; j++)
        {
            if (i != 0 && i != 3 && j != 0 && j != 3)
                cout << " ";
            else
                cout << "*";
        }
        cout << endl;
    }
}
void three()
{
    for (int i = 0; i < 4; i++)
    {
        for (int j = 0; j < i; j++)
            cout << " ";
        for (int j = 0; j < 4; j++)
        {
            cout << "*";
        }
        cout << endl;
    }
}
void four()
{
    for (int i = 0; i < 4; i++)
    {
        for (int j = 0; j < 8; j++)
        {
            cout << "*";
        }
        cout << endl;
    }
}
void five()
{
    for (int i = 0; i < 4; i++)
    {
        for (int j = 0; j < 8; j++)
        {
            if (i != 0 && i != 3 && j != 0 && j != 7)
                cout << " ";
            else
                cout << "*";
        }
        cout << endl;
    }
}
void six()
{
    for (int i = 0; i < 10; i++)
    {

        for (int j = 0; j < i; j++)
        {
            cout << " ";
        }
        for (int j = 0; j < 20; j++)
        {
            cout << "*";
        }
        cout << endl;
    }
}
void seven()
{

    for (int i = 0; i < 10; i++)
    {
        for (int j = 0; j < 10 - i; j++)
        {
            cout << " ";
        }
        for (int k = 0; k < 10; k++)
        {
            cout << "*";
        }
        cout << endl;
    }
}
void eight()
{
    for (int i = 0; i < 10; i++)
    {
        for (int j = 0; j < i; j++)
        {
            cout << "*";
        }
        cout << endl;
    }
}
void nine()
{
    for (int i = 0; i < 10; i++)
    {
        for (int j = 0; j < 10 - i; j++)
        {
            cout << " ";
        }
        for (int j = 0; j < 2 * i + 1; j++)
        {
            cout << "*";
        }
        cout << endl;
    }
}
void ten()
{
    for (int i = 0; i < 10; i++)
    {
        for (int j = 0; j < 10 - i; j++)
        {
            cout << " ";
        }
        for (int j = 0; j < 2 * i + 1; j++)
        {

            if (i == 0 || i == 9)
                cout << "*";
            else
            {
                if (j == 0 || j == 2 * i)
                    cout << "*";
                else
                    cout << " ";
            }
        }
        cout << endl;
    }
}
void eleven()
{
    for (int i = 0; i < 10; i++)
    {
        for (int j = 0; j < i; j++)
            cout << " ";
        for (int j = 0; j < 2 * (10 - i) - 1; j += 1)
        {
            cout << "*";
        }
        cout << endl;
    }
}
void twelve()
{
    for (int i = 0; i <= 10; i++)
    {
        for (int j = 0; j < i; j++)
        {
            cout << " ";
        }
        for (int j = 0; j < 2 * (10 - i) + 1; j++)
        {
            if (i == 0 || j == 0 || j == 2 * (10 - i))
            {
                cout << "*";
            }
            else
                cout << " ";
        }
        cout << endl;
    }
}
void thirteen()
{
    for (int i = 0; i < 8; i++)
    {
        if (i < 4)
        {
            for (int j = 0; j < i; j++)
            {
                cout << "*";
            }
        }
        else
        {
            for (int j = 0; j < 8 - i; j++)
            {
                cout << "*";
            }
        }
        cout << endl;
    }
}
void fourteen()
{
    for (int i = 0; i < 10; i++)
    {
        if (i < 5)
        {
            for (int j = 0; j < 10 - i; j++)
            {
                cout << " ";
            }
            for (int j = 0; j < i; j++)
            {
                cout << "*";
            }
        }
        else
        {
            for (int j = 0; j < i; j++)
            {
                cout << " ";
            }
            for (int j = 0; j < 10 - i; j++)
            {
                cout << "*";
            }
        }
        cout << endl;
    }
}
void fifteen()
{
    for (int j = 0; j < 20; j++)
    {
        if (j < 10)
        {
            for (int i = 0; i < 20 - j; i++)
            {
                cout << " ";
            }
            for (int i = 0; i < 2 * j + 1; i++)
            {
                cout << "*";
            }
        }
        else
        {
            for (int i = 0; i <= j; i++)
            {
                cout << " ";
            }
            for (int i = 0; i < 40 - (2 * j) - 1; i++)
            {
                cout << "*";
            }
        }

        cout << endl;
    }
}
int main()
{
    fifteen();
    return 0;
}